cdap-common/src/main/java/io/cdap/cdap/common/auditlogging/AuditLogSetterHook.java

```diff
@@ -68,19 +64,14 @@ public void postCall(HttpRequest request, HttpResponseStatus status, HandlerInfo
     long startTime = endTime - (endTimeNanos - startTimeNanos);
 
     LOG.trace("Setting a Audit Log event to published in SecurityRequestContext");
-    SecurityRequestContext.setAuditLogRequest(
-      new AuditLogRequest(
-        status.code(),
-        SecurityRequestContext.getUserIp(),
-        request.uri(),
-        getSimpleName(handlerInfo.getHandlerName()),
-        handlerInfo.getMethodName(),
-        request.method().name(),
-        SecurityRequestContext.getAuditLogQueue(),
-        startTime,
-        endTime
-      )
-    );
+    SecurityRequestContext.setAuditMetaDataInMap(AuditLogRequest.PropKey.OP_RESP_CODE, status.code());
```

Comment:
Could you please create a builder class for AutditLogRequest and use it here? It would be much better than using a `Map`. You can have different methods set different fields in the builder and later you just call build.